coffee-rest/src/main/java/hu/icellmobilsoft/coffee/rest/exception/DefaultBaseExceptionMapper.java

```diff
@@ -0,0 +1,160 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.rest.exception;
+
+import java.util.Collection;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+
+import org.apache.deltaspike.core.util.CollectionUtils;
+
+import hu.icellmobilsoft.coffee.cdi.logger.AppLogger;
+import hu.icellmobilsoft.coffee.cdi.logger.ThisLogger;
+import hu.icellmobilsoft.coffee.dto.common.commonservice.BONotFound;
+import hu.icellmobilsoft.coffee.dto.common.commonservice.BaseExceptionResultType;
+import hu.icellmobilsoft.coffee.dto.common.commonservice.BusinessFault;
+import hu.icellmobilsoft.coffee.dto.common.commonservice.InvalidRequestFault;
+import hu.icellmobilsoft.coffee.dto.common.commonservice.TechnicalFault;
+import hu.icellmobilsoft.coffee.dto.common.commonservice.ValidationType;
+import hu.icellmobilsoft.coffee.dto.exception.AccessDeniedException;
+import hu.icellmobilsoft.coffee.dto.exception.BONotFoundException;
+import hu.icellmobilsoft.coffee.dto.exception.BaseException;
+import hu.icellmobilsoft.coffee.dto.exception.BusinessException;
+import hu.icellmobilsoft.coffee.dto.exception.DtoConversionException;
+import hu.icellmobilsoft.coffee.dto.exception.InvalidRequestException;
+import hu.icellmobilsoft.coffee.dto.exception.ServiceUnavailableException;
+import hu.icellmobilsoft.coffee.dto.exception.XMLValidationError;
+import hu.icellmobilsoft.coffee.rest.validation.xml.exception.XsdProcessingException;
+
+/**
+ * Exception mapper for handled exception throwing
+ * 
+ * @author balazs.joo
+ * @since 1.8.0
+ */
+public class DefaultBaseExceptionMapper implements ExceptionMapper<BaseException> {
+
+    @Inject
+    @ThisLogger
+    private AppLogger log;
+
+    @Inject
+    private IExceptionMessageTranslator exceptionMessageTranslator;
+
+    /** {@inheritDoc} */
+    @Override
+    public final Response toResponse(BaseException e) {
+        log.error("Known error: ", e);
+        log.writeLogToError();
+
+        return handleException(e);
+    }
+
+    /**
+     * Kivétel kezelése
+     *
+     * @param e
+     *            a kivétel
+     * @return összeállított válasz
+     */
+    protected Response handleException(BaseException e) {
+        if (e instanceof AccessDeniedException) {
+            return createResponse(e, Response.Status.UNAUTHORIZED, new BusinessFault());
+        } else if (e instanceof BONotFoundException) {
+            return createResponse(e, IExceptionMessageTranslator.HTTP_STATUS_I_AM_A_TEAPOT, new BONotFound());
+        } else if (e instanceof DtoConversionException) {
+            return createResponse(e, Response.Status.BAD_REQUEST, new BusinessFault());
+        } else if (e instanceof InvalidRequestException) {
```

Comment:
Ez mar a 1.2.0 verzio ota deprecated, itt az ideje inkabb ki is szedni a kodbol komplett